pkg/ext/extension_apiserver.go

```diff
@@ -277,5 +281,55 @@ func NewExtensionAPIServer(ctx context.Context, wranglerContext *wrangler.Contex
 		return nil, fmt.Errorf("failed to install stores: %w", err)
 	}
 
+	// Note to self: the extension api server is handed to the caller to be
+	// run, which is done by handing it to a steveserver constructor. Not
+	// having direct access to the `Run` a goro with a delay is used to wait
+	// for the start and then perform the necessary factory action for
+	// ext'ension controllers.
+
+	go func() {
+		time.Sleep(3 * time.Second)
```

Comment:
why 3 seconds? could it take longer and fail? Can we somehow get rid of the sleep here?